tasks/component-builder.js

```diff
@@ -104,81 +42,115 @@ async function extractProperties(
 async function extractModifiers(filepath, { cwd } = {}) {
 	if (!fs.existsSync(filepath)) return Promise.resolve();
 
+	const componentName = cwd.split(path.sep).pop();
 	const content = await fsp.readFile(filepath, { encoding: "utf-8" });
 
 	/* Remove duplicates using a Set and sort the results (default is alphabetical) */
-	const found = await extractProperties(content);
-	const spectrum = await extractProperties(content, /(--spectrum-(?:\w|-)+)(?!:|\w|-)/g);
-	const system = await extractProperties(content, /(--system-(?:\w|-)+)(?!:|\w|-)/g);
-	const highContrast = await extractProperties(content, /(--highcontrast-(?:\w|-)+)(?!:|\w|-)/g);
+	const meta = extractProperties(content, {
+		modifiers: ["mod"],
+		spectrum: ["spectrum"],
+		"theming-layer": ["system"],
+		"high-contrast": ["highcontrast"],
+	});
 
+	const baseSelectors = [".spectrum", ".spectrum--express", ".spectrum--legacy"];
 	const selectors = new Set();
 	const root = postcss.parse(content);
-	root.walkRules(rule => {
-		if (rule.selector) selectors.add(rule.selector);
+	root.walkRules((rule) => {
+		if (rule.selectors) {
+			rule.selectors.forEach((selector) => {
+				// If the selector is not a base selector, add it to the set
+				if (baseSelectors.some((base) => selector !== base)) {
```

Comment:
I can see that this [isn't excluding selectors](https://github.com/adobe/spectrum-css/blob/ab74143c4d5ca11fa3cbb156cf9d07a972311766/components/actionbutton/metadata/metadata.json#L4) like `.spectrum`, should it be?

Reply:
Hmm, good question. I'm torn because they are selectors that are present in the file so it's accurate that they're listed but I'm not sure how helpful that inclusion is. What are your thoughts?

Reply:
I think that's ok if they're listed! Moreso I was just confused by the comment here, I was expecting it not to include those `baseSelectors`.

Reply:
Oh!  I was using a some selector instead of every when checking for !== so it wasn't excluding them like the code expected.  I fixed it!